test/new-e2e/system-probe/test-json-review/main.go

```diff
@@ -90,26 +49,51 @@ type testEvent struct {
 	Output  string
 }
 
+func testKey(test, pkg string) string {
+	return fmt.Sprintf("%s/%s", test, pkg)
+}
+
 func reviewTests(jsonFile string) (string, error) {
 	var failedTests strings.Builder
 	jf, err := os.Open(jsonFile)
 	if err != nil {
 		return "", fmt.Errorf("open %s: %s", jsonFile, err)
 	}
+	defer jf.Close()
 
 	scanner := bufio.NewScanner(jf)
+	testResults := make(map[string]testEvent)
 	for scanner.Scan() {
 		var ev testEvent
 		data := scanner.Bytes()
 		if err := json.Unmarshal(data, &ev); err != nil {
 			return "", fmt.Errorf("json unmarshal `%s`: %s", string(data), err)
 		}
-		if ev.Action == "fail" && ev.Test != "" {
-			failedTests.WriteString(fmt.Sprintf("FAIL: %s %s\n", ev.Package, ev.Test))
+		if ev.Test == "" {
+			continue
+		}
+		if res, ok := testResults[testKey(ev.Test, ev.Package)]; ok {
+			// If the test is already recorded as passed, it means the test
+			// eventually succeeded.
+			if res.Action == "pass" {
+				continue
+			}
+			if res.Action == "fail" && ev.Action == "pass" {
+				testResults[testKey(ev.Test, ev.Package)] = ev
```

Comment:
Do we want to record/output flaky tests separately?

Reply:
They will be made available to the CI visibility backend. Not sure how useful reporting them in the job output is...